src/common/src/util/memcmp_encoding.rs

```diff
@@ -165,6 +166,7 @@ fn calculate_encoded_size_inner(
             DataType::Jsonb => deserializer.skip_bytes()?,
             DataType::Varchar => deserializer.skip_bytes()?,
             DataType::Bytea => deserializer.skip_bytes()?,
+            DataType::Int256 => Int256::MEMCMP_ENCODED_SIZE,
```

Comment:
This is still incorrect. Although `deserializer.skip_bytes()` returns `Int256::MEMCMP_ENCODED_SIZE` here, it also has a side effect of advancing the offset. For fixed size types, the offset advancing is done on line 175 below.

Since we encoded int256 as bytes, it has extra marks and advancing 32 bytes is not enough.. But I am not familiar with how this code is used and how to construct a query to trigger this error or panic.

Now talking about the fix:
* Easiest is just to use `skip_bytes`, because the encoding is bytes.
* But actually we could avoid the extra marks in bytes encoding, and use exactly 32 bytes. There is `i256::into_words`. It returns `(i128, i128)` but logically it should be `(i128, u128)`. The memcomparable ser/de is not implemented for i128/u128 yet, but we can play the same trick again to [split i128](https://github.com/risingwavelabs/risingwave/blob/83a66ac6e30e1d2732871dd02f154e3fb1dde887/src/common/src/types/interval.rs#L734) to `(i64, u64)` and u128 to `(u64, u64)`. Let me enhance memcomparable for i128/u128 at the same time, which I should have done weeks earlier ...

Reply:
I'm not quite sure about code here, so let's change it to `skip_bytes` for now.